geoindicators/src/main/groovy/org/orbisgis/geoclimate/geoindicators/WorkflowGeoIndicators.groovy

```diff
@@ -1035,430 +1279,110 @@ IProcess computeAllGeoIndicators() {
+
+                def buildingTableName
+                def rsuTable
+                def buildingIndicatorsForHeightEst
+                def blockIndicatorsForHeightEst
+                def rsuIndicatorsForHeightEst
+                def nbBuildingEstimated
+
+                // Estimate building height
+                IProcess estimHeight = estimateBuildingHeight()
+                if (!estimHeight.execute(datasource: datasource, zoneTable: zoneTable, buildingTable: buildingTable,
+                                            roadTable: roadTable, railTable: railTable, vegetationTable: vegetationTable,
+                                            hydrographicTable: hydrographicTable, imperviousTable: imperviousTable,
+                                            buildingEstimateTableName: buildingEstimateTableName,
+                                            seaLandMaskTableName: seaLandMaskTableName,
+                                            surface_vegetation: surface_vegetation, surface_hydro: surface_hydro,
+                                            snappingTolerance: snappingTolerance, prefixName: prefixName,
+                                            buildingHeightModelName: buildingHeightModelName)) {
+                    error "Cannot estimate building height"
                     return
+                } else {
+                    buildingTableName = estimHeight.getResults().building_table_name
+                    rsuTable = estimHeight.getResults().rsu_table
+                    buildingIndicatorsForHeightEst = estimHeight.getResults().building_indicators_without_height
+                    blockIndicatorsForHeightEst = estimHeight.getResults().block_indicators_without_height
+                    rsuIndicatorsForHeightEst = estimHeight.getResults().rsu_indicators_without_height
+                    nbBuildingEstimated = estimHeight.getResults().nb_building_estimated
+                }
+
+                //This is a shortcut to extract building with estimated height
+                if (indicatorUse.isEmpty()) {
+                    //Clean the System properties that stores intermediate table names
+                    clearTablesCache()
+                    return [building_indicators: buildingIndicatorsForHeightEst,
+                            block_indicators   : blockIndicatorsForHeightEst,
+                            rsu_indicators     : rsuIndicatorsForHeightEst,
+                            rsu_lcz            : null,
+                            zone               : zoneTable,
+                            rsu_utrf_area      : null,
+                            rsu_utrf_floor_area: null,
+                            building_utrf      : null,
+                            building           : buildingTableName]
                 }
+
+                //The spatial relation tables RSU and BLOCK  must be filtered to keep only necessary columns
+                def rsuRelationFiltered = prefix prefixName, "RSU_RELATION_"
```

Comment:
Not sure to understand this filter

Reply:
rsuRelationFiltered used intead of rsuTable
If yes is rsuTable deleted ?

Reply:
This part of the code was already existing in the previous version. Not sure what contains rsuTable but it might be useless. Yes rsuTable is deleted.